feature: added check related functions

HannesSttzr opened the pull request.

Commit:
feature: added check related functions

HannesSttzr commented:
@v411e I changed the method parameter of the methods getAllowedMoves(), getAllowedFullMoves(), concretise() to use the Chessboard instead of the ChessGame, because they only used the ChessGame to get the Chessboard. But now there is a problem with the Canon because it uses the game to get the active player. Is there another way to get the active player (maybe by using a method like this ChessgameService.getActivePlayer())? What do you think? If we stick to the implementation with the ChessGame we have to pass the ChessGame to the ChessboardService. That I want to avoid.

Commit:
refactor: resolved some code smells

Commit:
refactor: updated the tests and reduced duplication

v411e commented:
> @v411e I changed the method parameter of the methods getAllowedMoves(), getAllowedFullMoves(), concretise() to use the Chessboard instead of the ChessGame, because they only used the ChessGame to get the Chessboard. But now there is a problem with the Canon because it uses the game to get the active player. Is there another way to get the active player (maybe by using a method like this ChessgameService.getActivePlayer())? What do you think? If we stick to the implementation with the ChessGame we have to pass the ChessGame to the ChessboardService. That I want to avoid.

Alright, thanks for the hint. I will push a fix to this branch.

v411e committed:
Merge branch 'develop' of github.com:ccd-course/chess_backend into feature/checkmate

# Conflicts:
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/Shoot.java
#	src/main/java/com/chess/backend/services/ChessboardService.java

v411e committed:
Bugfixes and Merge branch 'develop' of github.com:ccd-course/chess_backend into feature/checkmate

# Conflicts:
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveBackward.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveDiagonal.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveForward.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveKnight.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveLeft.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveOneBackward.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveOneDiagonal.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveOneForward.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveOneLeft.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveOneRight.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MovePawnCaptureBackward.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MovePawnCaptureForward.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveRight.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveTwoBackward.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/MoveTwoForward.java
#	src/main/java/com/chess/backend/gamemodel/abstractmoves/Shoot.java
#	src/main/java/com/chess/backend/gamemodel/pieces/Bishop.java
#	src/main/java/com/chess/backend/gamemodel/pieces/Cannon.java
#	src/main/java/com/chess/backend/gamemodel/pieces/Ferz.java
#	src/main/java/com/chess/backend/gamemodel/pieces/King.java
#	src/main/java/com/chess/backend/gamemodel/pieces/Knight.java
#	src/main/java/com/chess/backend/gamemodel/pieces/Pawn.java
#	src/main/java/com/chess/backend/gamemodel/pieces/Queen.java
#	src/main/java/com/chess/backend/gamemodel/pieces/Rook.java
#	src/main/java/com/chess/backend/gamemodel/pieces/Wazir.java
#	src/main/java/com/chess/backend/services/ChessGameService.java
#	src/main/java/com/chess/backend/services/ChessboardService.java
#	src/test/java/com/chess/backend/AbstractMovementTests.java
